lib/models/device.ts

```diff
@@ -2998,6 +3005,25 @@ const getDeviceModel = function (
 		): Promise<void> => {
 			let deviceId;
 			let releaseId;
+			const deviceOpts = {
+				$select: toWritable([
+					'id',
+					'supervisor_version',
+					'os_version',
+				] as const),
+				$expand: { is_of__device_type: { $select: 'slug' } },
+			} as const;
+
+			const device = (await exports.get(
+				uuidOrId,
+				deviceOpts,
+			)) as PineTypedResult<Device, typeof deviceOpts>;
+			ensureVersionCompatibility(
+				device.supervisor_version,
+				MIN_SUPERVISOR_MC_API,
+				'supervisor',
+			);
+			ensureVersionCompatibility(device.os_version, MIN_OS_MC, 'host OS');
```

Comment:
Does the API do any such check as well?
I'm not against this, since it gives a better error message, but having this in a hook of the API might be better.

Reply:
the API does not enforce this at the moment, as it felt a little overkill for a small edge case. i can enforce it as part of a hook if you'd be more comfortable enforcing it in both places.

Reply:
thinking through this, we can't enforce this at the API level since devices that are being HUPed still require supervisor updates. we could potentially limit it to only device API keys, but i'm sure if that complexity is worth it.